tests/fixtures/application_fixtures.py

```diff
@@ -155,6 +160,39 @@ def mock_upload_logic() -> Generator:
         yield _fixture
 
 
+@pytest.fixture(scope="function", autouse=True)
```

Comment:
I'm going to watch the CI time here, since this effectively tears down and re-adds the data categories before and after each test is run.